presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1369,4 +1349,46 @@ public ActualProperties getProperties()
             return properties;
         }
     }
+
+    private class LayoutConstraintEvaluator
+    {
+        private final List<ExpressionInterpreter> evaluators;
+        private final Map<Symbol, ColumnHandle> assignments;
+
+        public LayoutConstraintEvaluator(Session session, Map<Symbol, Type> types, Map<Symbol, ColumnHandle> assignments, List<Expression> conjuncts)
+        {
+            this.assignments = assignments;
+
+            ImmutableList.Builder<ExpressionInterpreter> builder = ImmutableList.builder();
+            for (Expression conjunct : conjuncts) {
```

Comment:
nit: Would it be possible to combine conjuncts into single expression (AND) and run `ExpressionInterpreter` on that?

Reply:
Good point. I was just working with the existing code and didn't stop to think why it was doing it this way.